varlink/connection.go

```diff
@@ -238,6 +245,24 @@ func (c *Connection) GetInfo(ctx context.Context, vendor *string, product *strin
 	return nil
 }
 
+// Upgrade attempts to upgrade the connection using the provided method and parameters.
+// If successful, the connection cannot be reused later, and must be closed.
+func (c *Connection) Upgrade(ctx context.Context, method string, parameters interface{}) (func(context.Context, interface{}) (uint64, ReadWriterContext, error), error) {
+	reply, err := c.Send(ctx, method, parameters, Upgrade)
+	if err != nil {
+		return nil, err
+	}
+
+	return func(ctx context.Context, out interface{}) (uint64, ReadWriterContext, error) {
+		flags, err := reply(ctx, out)
+		if err != nil {
+			return 0, nil, err
+		}
+
+		return flags, c.conn, nil
```

Comment:
what is the intended usage of `flags`?

Reply:
It is exposed at this level so that generated code can access it. Hiding this information may make the function impossible to use for certain use cases.